### ISSUE TITLE
WCAG 2 should be clearer that for responsive sites, each breakpoint/view needs to be conforming, and different breakpoints can't be considered conforming alternatives

### ISSUE THREAD

**Issue by DavidMacDonald** [w3c/wcag21#19]
Filing this here in addition to the WCAG 2 Repo https://github.com/w3c/wcag/issues/197. 
As a fallback in case some feel it shouldn't be amended in 2.0... it is to clarify that a "Web Page" in WCAG means that every break point of that page needs to conform.

There was quite a long thread on the list, and this is the outcome of the discussion. I presented both Gregg and Loretta with this scenario

> Company X has a responsive web site. A page has 2 break points based on viewport size. A user on a mobile device gets the same page as as the desktop, except it has a Hamburger menu icon instead of the mega menu.
> 
> The mega menu conforms to WCAG, the Hamburger menu does not. On a mobile device, there is no link to the desktop version. Does this page conform to WCAG?
> 
> Some feel that it currently passes because there is one accessibility supported solution. Others think that it does not pass because the user on the mobile device doesn't have a choice about which view they get, (unless there is an accessible link to the desktop -alternative conforming- version.)
> 
> What are your thoughts? 
> 
> 1) Does it pass without a link to the conforming desktop view of the same page?
> 2) Does it pass if it has a link to the conforming desktop view of the same page
> 
> The question is important for decisions in 2.1

Loretta Answers: 

> Just a quick response because I'm on vacation and can't investigate the historical record.
> 
> We had a lot if relevant discussion when trying to define web page (or all the alternate terms we were considering). Part of the discussion was about web apps, which dynamically change what is delivered from the same URI. As I recall, we concluded that any version of the page that could be delivered from that URI had to conform in order to claim that the URI conformed.
> 
> I think this generalizes to saying that all the break points must conform to claim that the URI conforms.
> 
> But this is just my recollection. Sorry I can't point to a specific discussion.

Gregg answers:

> I would think about it like I would a site where there was an accessible alternative - but the alternative was behind a firewall and only available to some. 
> 
> Unless the accessible version is available to all - it does not qualify and an alternative for all (and fails the SC).
> 
> They essentially have a page that morphs into an inaccessible form that the user has no control of.
> 
> IN SHORT
> - it fails the conformance clause because the accessible alternative  is only available from the in accessible version under certain conditions - not at all time. 

Most others agree also. The clarification that a "full page" in Conformance Criteria 2 includes each of it's breakpoints, does not appear anywhere in the current documentation. I think in an age where most sites are responsive this is necessary. There is no proposal to change WCAG, just a statement in the Understanding Conformance Criteria 2 section about this: The proposal is a follows:

> "The full page includes each view of the page that is customized for various devices, browsers, or screen sizes. Each of these views (or their respective conforming alternate versions) would need to conform in order for the entire page to conform. On the other hand, if a user voluntarily chooses a setting on the page that optimizes or personalizes the state of the page for accessibility reasons, this new state does not necessarily need to pass every Success Criteria in order to personalize the page, because the conforming version can be reached by undoing the setting."

It would be placed after the last paragraph in the section "Understanding Requirement 2" (just before the Notes at the end of the section. https://www.w3.org/TR/UNDERSTANDING-WCAG20/conformance.html#uc-conformance-requirements-head

**Comment by awkawk** [comment]
The WCAG21 issues area is for new SC proposals only. If you logged this elsewhere, great, closing it here.

**Comment by DavidMacDonald** [comment]
@awkawk 
 It appears to be closed both in WCAG 2 https://github.com/w3c/wcag/issues/197

I don't care where it is open, either place is ok with me, but it needs to be open and discussed by the group before being closed, I would say.

**Comment by DavidMacDonald** [comment]
I'm requesting that this issue be reopened because I've closed it over on the WCAG 2.0 issue list.
There is enough ambiguity over whether WCAG 2 applies to each break point that I think we'll need to take it up as a new issue.
https://github.com/w3c/wcag/issues/197

**Comment by patrickhlauke** [comment]
while the reasoning in here and in https://github.com/w3c/wcag/issues/197 is sound, just a note on the title of this issue: it gives a false impression, as there's not necessarily a distinction between "desktop" and "responsive" here, i.e. the responsive layout will adapt to different sizes, so it's "in effect" when you're on desktop too.

**Comment by patrickhlauke** [comment]
(sorry, amended it to be a bit long...but hopefully unambiguous)

**Comment by DavidMacDonald** [comment]
Gregg has brought up the possibility of specialized delivery options of pages such as an earpiece. Although speculative, perhaps there is a better to limit the amendment to screen sizes. 

"The full page includes each variation of the page that is automatically generated by the page for various screen sizes. Each of these variations (or their respective conforming alternate versions) needs to conform in order for the entire page to conform. However, if a user actively chooses a setting on the page that optimizes or personalizes the state of the page for accessibility reasons, this new state does not necessarily need to conform, because the conforming version can be reached by undoing the setting."

**Comment by awkawk** [comment]
Addressed in conformance criteria change
